mockgcp/mockcompute/nodegroupsv1.go

```diff
@@ -0,0 +1,183 @@
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node group %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading node group: %v", err)
+		}
+	}
+
+	return obj, nil
+}
+
+func (s *NodeGroupsV1) Insert(ctx context.Context, req *pb.InsertNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroupResource().GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	id := s.generateID()
+
+	obj := proto.Clone(req.GetNodeGroupResource()).(*pb.NodeGroup)
+	obj.SelfLink = PtrTo("https://compute.googleapis.com/compute/v1/" + name.String())
+	obj.CreationTimestamp = PtrTo(s.nowString())
+	obj.Id = &id
+	obj.Kind = PtrTo("compute#nodegroup")
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating node group: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) Patch(ctx context.Context, req *pb.PatchNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.NodeGroup{}
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating node group: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) Delete(ctx context.Context, req *pb.DeleteNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeGroup{}
+	if err := s.storage.Delete(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node group %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error deleting node group: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *NodeGroupsV1) SetNodeTemplate(ctx context.Context, req *pb.SetNodeTemplateNodeGroupRequest) (*pb.Operation, error) {
+	name, err := s.newNodeTemplateNodeGroupName(req.GetProject(), req.GetZone(), req.GetNodeGroup(), req.GetNodeGroupsSetNodeTemplateRequestResource().GetNodeTemplate())
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.NodeGroup{}
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "node template node group %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading node template node group: %v", err)
+		}
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+type nodeGroupName struct {
+	Project *projects.ProjectData
+	Zone    string
+	Name    string
+}
+
+type nodeTemplateNodeGroupName struct {
```

Comment:
I don't think you need this (assuming setNodeTemplate maps to a field)

Reply:
removed